packages/libs/components/src/components/plotControls/TimeSlider.tsx

```diff
@@ -96,9 +96,35 @@ function TimeSlider(props: TimeSliderProps) {
       debounce((domain: Bounds | null) => {
         if (!domain) return;
         const { x0, x1 } = domain;
+
+        // check initial range in pixels: ideally it should be [0, 700] (pixels),
+        // but there exists offset about 2 pixel, i.e., [-2, 702]
+        console.log(`start: ${xBrushScale(x0)}, end: ${xBrushScale(x1)}`);
+
+        // computing offset of 2 pixel in domain (milliseconds)
+        // for the example of Geolocation visualizations of field-based studies
+        // and 'species collection start date' variable,
+        // such a 2 pixel results in about 4 months
+        const brushOffset =
+          xBrushScale.invert(2).getTime() - xBrushScale.domain()[0].getTime();
```

Comment:
I prefer this for readability and also just in case the domain doesn't start at zero one day!
I've tested it and it works fine.
```suggestion
          xBrushScale.invert(2).getTime() - xBrushScale.invert(0).getTime();
```

Reply:
Oh, and we should add a comment referring to the SAFE_PIXEL constant in the visx source code
https://github.com/airbnb/visx/blob/86a851cb3bf622b013b186f02f955bcd6548a87f/packages/visx-brush/src/Brush.tsx#L14 

And remove the console.logs before merging. Thanks!

Reply:
@bobular Thank you for your tests and reviews 👍 Sounds great it works! I followed your comments and made a new commit. Also, changed this PR to a regular one.